### PR TITLE
[train] minor documentation improvements

### PR DESCRIPTION
**Pull request by matthewdeng**
## Why are these changes needed?

1. Add images to `More Frameworks`.
2. Remove `Transformers` section from `More Frameworks` since there is now a dedicated page.
3. Cleanup API Reference page.
    1. Reorder Train Loop APIs in general usage.
    2. Remove Restore API references since they're already part of the Trainer APIs
    3. Remove BaseTrainer methods because they're already in BaseTrainer
    4. Add additional APIs like `Result` and `FailureConfig`.
         1. **Note:** This makes the AIR APIs fail to generate, but I think this is okay given that we are moving all code to point to the train counterparts, and will be removing the air API references in 2.7.
 4. Update ToC.
     1. Remove "Ray Train" prefix for some pages.
     2. Reorder pages so Examples come before Internals.
     3. Remove unneeded dropdown icon for APIs. 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/train/api/api.rst [hunk 1]
```diff
@@ -35,13 +35,13 @@ PyTorch
 .. autosummary::
     :toctree: doc/
 
+    ~train.torch.get_device
```
Comment (woshiyyya) [inline doc/source/train/api/api.rst:38]: I feel there are too many apis here, can we separate them into basic(`get_device`, `prepare_model`, `prepare_data_loader`) and advanced(others)?

File: doc/BUILD [hunk 2]
```diff
@@ -223,9 +223,7 @@ py_test(
 py_test_run_all_subdirectory(
     size = "large",
     include = ["source/train/doc_code/*.py"],
-    exclude = [
-        "source/train/doc_code/hf_trainer.py",  # Too large
-    ],
+    exclude = [],
```
Comment (woshiyyya) [inline doc/BUILD:226]: why are we adding this test back?

**Review by woshiyyya — APPROVED** [review]
Thanks @matthewdeng ! Looks good to me. Left 2 minor comments.

File: doc/BUILD (hunk 2 above)
Comment (matthewdeng) [reply doc/BUILD:226]: Oops, it's because I got rid of the section that uses it. I forgot to remove the file.

**Review by richardliaw — APPROVED** [review]

**Pull request merged**